scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/simulation/InMemoryCachedEngineStoreTest.java

```diff
@@ -0,0 +1,110 @@
+        Duration.SECOND,
+        Map.of(
+            new ActivityDirectiveId(1), new ActivityDirective(Duration.HOUR, "ActivityType1", Map.of(), null, true),
+            new ActivityDirectiveId(2), new ActivityDirective(Duration.HOUR, "ActivityType2", Map.of(), null, true)
+        ),
+        new SimulationEngine(),
+        new LiveCells(new CausalEventSource()),
+        new SlabList<>(),
+        null,
+        SimulationUtility.getFooMissionModel()
+    );
+  }
+
+  public static CheckpointSimulationDriver.CachedSimulationEngine getCachedEngine2(){
+    return new CheckpointSimulationDriver.CachedSimulationEngine(
+        Duration.SECOND,
+        Map.of(
+            new ActivityDirectiveId(3), new ActivityDirective(Duration.HOUR, "ActivityType3", Map.of(), null, true),
+            new ActivityDirectiveId(4), new ActivityDirective(Duration.HOUR, "ActivityType4", Map.of(), null, true)
+        ),
+        new SimulationEngine(),
+        new LiveCells(new CausalEventSource()),
+        new SlabList<>(),
+        null,
+        SimulationUtility.getFooMissionModel()
+    );
+  }
+
+  public static CheckpointSimulationDriver.CachedSimulationEngine getCachedEngine3(){
+    return new CheckpointSimulationDriver.CachedSimulationEngine(
+        Duration.SECOND,
+        Map.of(
+            new ActivityDirectiveId(5), new ActivityDirective(Duration.HOUR, "ActivityType5", Map.of(), null, true),
+            new ActivityDirectiveId(6), new ActivityDirective(Duration.HOUR, "ActivityType6", Map.of(), null, true)
+        ),
+        new SimulationEngine(),
+        new LiveCells(new CausalEventSource()),
+        new SlabList<>(),
+        null,
+        SimulationUtility.getFooMissionModel()
+    );
+  }
+
+  @Test
+  public void duplicateTest(){
+    final var store = new InMemoryCachedEngineStore(2);
+    store.save(CheckpointSimulationDriver.CachedSimulationEngine.empty(SimulationUtility.getFooMissionModel()), this.simulationEngineConfiguration);
+    store.save(CheckpointSimulationDriver.CachedSimulationEngine.empty(SimulationUtility.getFooMissionModel()), this.simulationEngineConfiguration);
+    store.save(CheckpointSimulationDriver.CachedSimulationEngine.empty(SimulationUtility.getFooMissionModel()), this.simulationEngineConfiguration);
+    assertEquals(1, store.getCachedEngines(this.simulationEngineConfiguration).size());
+  }
+
+  @Test
+  public void order(){
+    final var store = new InMemoryCachedEngineStore(2);
+    final var cachedEngine1 = getCachedEngine1();
+    final var cachedEngine2 = getCachedEngine2();
+    final var cachedEngine3 = getCachedEngine3();
+    store.save(cachedEngine1, this.simulationEngineConfiguration);
+    store.save(cachedEngine2, this.simulationEngineConfiguration);
+    final var cachedBeforeRegister = store.getCachedEngines(this.simulationEngineConfiguration);
+    //engines have 0 used, so they are ordered in descending creation date
```

Comment:
Minor, non-blocking: Clarity
```suggestion
    // no engines have been used, so the cache is ordered in descending creation date
```